### ISSUE TITLE
Cannot create an app with the same name as a previously deleted app

### ISSUE THREAD

**Issue by devj3ns** [firebase/firebase-js-sdk#5464]
This issue is related to this flutter-fire issue: https://github.com/FirebaseExtended/flutterfire/issues/4454.

I wanted to create a new user without getting logged out from my current account. Unfortuantely I got the following error when trying to delete the created second app again (which I used to create the new user):

```dart
Expected a value of type 'JSObject<I>', but got one of type 'NativeJavaScriptObject'
```

On android it worked fine.

---

As @russellwheatley said in the issue linked above 'This is the behaviour of the firebase-js-sdk, and it appears you cannot create an app with the same name as a previously deleted app.'

He used this code to recreate the error on the web platform:

```dart
var firebaseConfig = {
  // config
};

// Initialize Firebase
firebase.initializeApp(firebaseConfig);

var second = firebase.initializeApp(firebaseConfig, 'secondApp');

var secondAuth = second.auth();

export async function called() {
  console.log('still here??');
  await secondAuth.createUserWithEmailAndPassword(
    `fake@email.io${Date.now()}`,
    'blahblah'
  );

  console.log('secondAuth.currentUser', secondAuth.currentUser);

  await secondAuth.signOut();

  await second.delete();
}

document.getElementById('button-id').onclick = async () => {
  await called(); 
};
```

It reproduces the exact same error received using flutter web:

```
FirebaseError: Firebase: Firebase App named 'secondApp' already deleted (app/app-deleted).
```

So there is an inconsistent behaviour to android and iOS.

**Comment by Feiyang1** [comment]
Interesting. And you are able to do so on Android and iOS?
As a workaround, you may keep a counter, or use a timestamp to append to the app name for the next app you are creating. This is definitely a valid use case where we can make it easier for you by allowing recreating a deleted app. I will take it to team and see if there is any concern.

Also, the example you gave doesn't manifest the issue - do you mean to recreate the second firebase app within `called()` where you also delete it at the end in the example?

**Comment by devj3ns** [comment]
Hey, yeah, on Android and iOS it is working.

I have not written the code above and not tested on native web with `firebase-js-sdk`. Only with flutter web. You can see the code I used [here](https://github.com/FirebaseExtended/flutterfire/issues/4454#issuecomment-755280822). What I am doing there is the following:

1. Create a second instance
2. Create the new account in that instance
3. Delete the instance
4. Create a second instance with the same instance again
5. Create a new account in that instance
6. Delete the instance

And somewhere in the last three steps the exception is thrown on the web.

**Comment by russellwheatley** [comment]
Hey @Feiyang1, this is reproducible on version `8.6.1` which Flutterfire uses (it's not possible to use version `>9.0.0` due to the nature of the imports used).  Presumably there will be no more releases for version 8?

**Comment by Feiyang1** [comment]
@russellwheatley Correct. You can use the v9 compat packages if you are stuck in the old API:
```
import firebase from 'firebase/compat/app';
import from 'firebase/compat/auth';
...
```

**Comment by Feiyang1** [comment]
@russellwheatley I couldn't reproduce the issue with `8.6.1`. `app2` is deleted and recreated successfully, code:
```javascript
import firebase from 'firebase/app';
import 'firebase/auth'

firebase.initializeApp({...});

async function createSecondApp() {
    const app2 = firebase.initializeApp({...}, 'app2');
    console.log('created app2');

    await app2.auth().signInAnonymously();

    await app2.delete();
}

createSecondApp().then(createSecondApp).then(createSecondApp)
```

Am I missing something?
